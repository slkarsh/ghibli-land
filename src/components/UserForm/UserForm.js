import React, { Component } from 'react'
import './UserForm.scss'
import { connect } from 'react-redux'
import { addTitleAndPlot } from '../../actions'

class UserForm extends Component {
  constructor() {
    super();
    this.state = {
      title: '',
      plot: '',
      score: null
    }
  }
  
  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  submitForm = e => {
    const { title, plot } = this.state
    const { diyMovie, addTitleAndPlot } = this.props
    console.log('made it to submit form')
    if (title === '' || plot === '' || diyMovie.characters.length === 0 || diyMovie.locations.length === 0 || diyMovie.vehicles.length === 0) {
      e.preventDefault()
      this.setState({score: 0})
      console.log('0', this.state.score)
    }
    else {
      console.log('helloooo')
      this.getScore()
      addTitleAndPlot(this.state)
    }
  }

  getScore = () => {
    console.log('get score was called')
    let randomScore =  Math.floor((Math.random() * 50))
    console.log('random score', randomScore)
    this.setState({score: randomScore})
    return randomScore
  }

  clearInputs = () => {
    this.setState({
      title: '',
      plot: '',
      score: null
    })
  }

  render() {
    return (
      <section className='form-section'>
        <form className='user-form'>
          <input
            name='title'
            value={this.state.title}
            onChange={this.handleChange}
            placeholder='Name your movie'
            className='title-input'
           />
           <textarea
            rows='30'
            columns='100'
            name='plot'
            value={this.state.plot}
            onChange={this.handleChange}
            placeholder='Describe the plot'
            className='plot-input'
           />
        </form>
        <button className='submit-button' onClick={(e) => this.submitForm(e)}>See Your Rotten Tomatoes Score!</button>
        {this.state.score && <div className='movie-score'>{this.state.score}</div>}
        {!this.state.score && <div className='movie-score'>Make sure you have added all the information to make a film!</div>}
      </section>
    )
  }
}

const mapStateToProps = state => ({
  diyMovie: state.diyMovie
})

const mapDispatchToProps = dispatch => ({
  addTitleAndPlot: (inputInfo) => dispatch(addTitleAndPlot(inputInfo))
})

export default connect(mapStateToProps, mapDispatchToProps)(UserForm)