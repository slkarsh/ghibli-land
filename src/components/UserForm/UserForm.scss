.form-section {
  margin: auto;
}

.user-form {
  display: flex;
  flex-direction: column;
  margin-top: 30px;

}

.title-input {
  height: 65px;
  margin-bottom: 20px;
  font-size: 1em;
}

.plot-input {
  height: 350px;
  overflow: scroll;
  font-size: 1em;
}

.submit-button {
  background-color: transparent;
  border: none;
  font-size: 1.2em;
  margin: 10px auto;
}

.movie-score {
  font-size: 2em;
  margin: auto;
  text-align: center;
}
