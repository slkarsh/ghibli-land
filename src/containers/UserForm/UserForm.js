import React, { Component } from 'react'
import './UserForm.scss'
import { connect } from 'react-redux'
import { addTitleAndPlot } from '../../actions'
import PropTypes from 'prop-types'

export class UserForm extends Component {
  constructor() {
    super();
    this.state = {
      title: '',
      plot: '',
      score: 0
    }
  }
  
  handleChange = e => {
    e.preventDefault()
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  submitForm = e => {
    const { title, plot } = this.state
    const { diyMovie, addTitleAndPlot } = this.props
    console.log('made it to submit form')
    if (title === '' || plot === '' || diyMovie.characters.length === 0 || diyMovie.locations.length === 0 || diyMovie.vehicles.length === 0) {
      e.preventDefault()
      this.setState({score: 0})
    }
    else {
      this.getScore()
      addTitleAndPlot(this.state)
      // this.clearInputs()
    }
  }

  getScore = () => {
    let randomScore =  Math.floor((Math.random() * 50))
    console.log('random score', randomScore)
    this.setState({score: randomScore})
    console.log('score', this.state.score)
    return randomScore
  }

  clearInputs = () => {
    this.setState({
      title: '',
      plot: '',
      score: 0
    })
  }

  render() {
    return (
      <section className='form-section'>
        <form className='user-form'>
          <input
            name='title'
            value={this.state.title}
            onChange={this.handleChange}
            placeholder='Name your movie'
            className='title-input'
           />
           <textarea
            rows='30'
            columns='100'
            name='plot'
            value={this.state.plot}
            onChange={this.handleChange}
            placeholder='Describe the plot'
            className='plot-input'
           />
        </form>
        <button className='submit-button' onClick={(e) => this.submitForm(e)}>See Your Rotten Tomatoes Score!</button>
        {this.state.score !== 0 && <div className='movie-score'>{this.state.score}</div>}
        {this.state.score === 0 && <div className='movie-score'>Make sure you have added all the information to make a film!</div>}
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  diyMovie: state.diyMovie
})

export const mapDispatchToProps = dispatch => ({
  addTitleAndPlot: (inputInfo) => dispatch(addTitleAndPlot(inputInfo))
})

export default connect(mapStateToProps, mapDispatchToProps)(UserForm)

UserForm.propTypes = {
  title: PropTypes.string,
  plot: PropTypes.string,
  diyMovie: PropTypes.object,
  addTitleAndPlot: PropTypes.func,
  submitForm: PropTypes.func,
  handleChange: PropTypes.func,
  getScore: PropTypes.func,
  clearInputs: PropTypes.func
}