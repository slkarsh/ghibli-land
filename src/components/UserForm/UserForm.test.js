import React from 'react'
import { shallow } from 'enzyme'
import { UserForm, mapStateToProps, mapDispatchToProps } from './UserForm'

describe('UserForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<UserForm />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should update the title input onChange', () => {
    const mockEvent = { preventDefault() { }, target: { name: 'title', value: 'some title' } };

    const expected = 'some title'

    wrapper.instance().handleChange(mockEvent)

    expect(wrapper.state('title')).toEqual(expected)
  })

  it('should update the plot input onChange', () => {
    const mockEvent = { preventDefault() { }, target: { name: 'plot', value: 'some plot' } };

    const expected = 'some plot'

    wrapper.instance().handleChange(mockEvent)

    expect(wrapper.state('plot')).toEqual(expected)
  })

  it('should call the submitForm method when the button is clicked', () => {
    wrapper.instance().submitForm = jest.fn()
    wrapper.instance().forceUpdate()

    wrapper.find('button').simulate('click')

    expect(wrapper.instance().submitForm).toHaveBeenCalled()
  })

  it.skip('should call the submitForm method with the properties in state when the button is clicked', () => {
    const mockTitle = 'a title'
    const mockPlot = 'a plot'

    wrapper.instance().submitForm = jest.fn()

    wrapper.instance().setState({
      title: mockTitle,
      plot: mockPlot
    })

    wrapper.find('button').simulate('click')

    expect(wrapper.instance().submitForm).toHaveBeenCalledWith(mockTitle, mockPlot)
  })
})