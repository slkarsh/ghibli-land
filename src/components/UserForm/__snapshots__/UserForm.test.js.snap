// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm should match the snapshot 1`] = `
<section
  className="form-section"
>
  <form
    className="user-form"
  >
    <input
      className="title-input"
      name="title"
      onChange={[Function]}
      placeholder="Name your movie"
      value=""
    />
    <textarea
      className="plot-input"
      columns="100"
      name="plot"
      onChange={[Function]}
      placeholder="Describe the plot"
      rows="30"
      value=""
    />
  </form>
  <button
    className="submit-button"
    onClick={[Function]}
  >
    See Your Rotten Tomatoes Score!
  </button>
  <div
    className="movie-score"
  >
    Make sure you have added all the information to make a film!
  </div>
</section>
`;
